locatr.go

```diff
@@ -66,16 +72,23 @@ type BaseLocatrOptions struct {
 	UseCache bool
 	// LogConfig is the log configuration
 	LogConfig LogConfig
+
+	// LocatrResultsFilePath is the path to the file where the locatr results will be written
+	// If not provided, the results will be written to DEFAULT_LOCATR_RESULTS_FILE
+	ResultsFilePath string
 }
 
 // NewBaseLocatr creates a new instance of BaseLocatr
 // plugin: (playwright, puppeteer, etc)
 // llmClient: struct that are returned by NewLlmClient
 // options: All the options for the locatr package
-func NewBaseLocatr(plugin PluginInterface, llmClient LlmClient, options BaseLocatrOptions) *BaseLocatr {
+func NewBaseLocatr(plugin PluginInterface, llmClient LlmClientInterface, options BaseLocatrOptions) *BaseLocatr {
 	if len(options.CachePath) == 0 {
 		options.CachePath = DEFAULT_CACHE_PATH
 	}
+	if len(options.ResultsFilePath) == 0 {
```

Comment:
let's make storing the results (in memory as well) optional. as done in caching. let's not write to memory if the flag for storing results isn't passed.